libs/sdk-ui-ext/src/internal/interfaces/VisualizationDescriptor.ts

```diff
@@ -0,0 +1,95 @@
+// (C) 2021 GoodData Corporation
+import { IInsightDefinition } from "@gooddata/sdk-model";
+import { IVisConstruct, IVisualization } from "./Visualization";
+
+/**
+ * Factories that create a new instance of pluggable visualization.
+ *
+ * @alpha
+ */
+export type PluggableVisualizationFactory = (params: IVisConstruct) => IVisualization;
+
+/**
+ * Info about min, max and default in layout units.
+ * If no value is provided for min/max/default layout will use its generic defaults for it.
+ *
+ * @alpha
+ */
+interface ISizeInfo {
+    default?: number;
+    min?: number;
+    max?: number;
+}
+
+/**
+ * Info about visualization's min, max and default for width and height.
+ *
+ * @alpha
+ */
+export interface IVisualizationSizeInfo {
+    width: ISizeInfo;
+    height: ISizeInfo;
+}
+
+/**
+ * Layout descriptor.
+ * Provides parameters of used layout.
+ * Width of the grid, height of one grid row and others
+ *
+ * @alpha
+ */
+export interface ILayoutDescriptor {
+    gridColumnsCount: number;
+    gridRowsCount: number;
+    gridColumnWidth: number;
+    gridRowHeight: number;
```

Comment:
it needs to be clear what these things mean. think about reading the comments and from this trying to figure out what values to return in size info...?